Libraries/ConfigParser/src/config_parser.cpp

```diff
@@ -1,247 +1,266 @@
+
+#define WIN32_LEAN_AND_MEAN // exclude rarely used windows content
+#include <windows.h>
+
+#include <third_party/jsoncpp/source/include/json/json.h>
+
+using namespace StreamingToolkit;
+
+const char* ConfigParser::kWebrtcConfigPath = "webrtcConfig.json";
+const char* ConfigParser::kServerConfigPath = "serverConfig.json";
+const char* ConfigParser::kNvEncConfigPath = "nvEncConfig.json";
+
+std::string ConfigParser::GetAbsolutePath(const std::string& file_name)
+{
+	TCHAR buffer[MAX_PATH];
+	GetModuleFileName(NULL, buffer, MAX_PATH);
+	std::string::size_type pos = std::string(buffer).find_last_of("\\/");
+	return std::string(buffer).substr(0, pos + 1) + file_name;
+}
+
+void ConfigParser::ConfigureConfigFactories()
+{
+	ConfigureConfigFactories(GetAbsolutePath(""));
+}
+
+void ConfigParser::ConfigureConfigFactories(const std::string& baseFilePath)
+{
+	CppFactory::Object<StreamingToolkit::WebRTCConfig>::RegisterAllocator([=] {
+		auto config = new StreamingToolkit::WebRTCConfig();
+
+		ConfigParser::ParseWebRTCConfig(baseFilePath + std::string(ConfigParser::kWebrtcConfigPath), config);
+
+		return std::shared_ptr<StreamingToolkit::WebRTCConfig>(config);
+	});
+
+	CppFactory::Object<StreamingToolkit::ServerConfig>::RegisterAllocator([=] {
+		auto config = new StreamingToolkit::ServerConfig();
+
+		ConfigParser::ParseServerConfig(baseFilePath + std::string(ConfigParser::kServerConfigPath), config);
+
+		return std::shared_ptr<StreamingToolkit::ServerConfig>(config);
+	});
+
+	CppFactory::Object<StreamingToolkit::FullServerConfig>::RegisterAllocator([=] {
+		auto config = new StreamingToolkit::FullServerConfig();
+
+		config->server_config = CppFactory::Object<StreamingToolkit::ServerConfig>::Get();
+		config->webrtc_config = CppFactory::Object<StreamingToolkit::WebRTCConfig>::Get();
+
+		return std::shared_ptr<StreamingToolkit::FullServerConfig>(config);
+	});
+
+	CppFactory::Object<StreamingToolkit::NvEncConfig>::RegisterAllocator([=] {
+		auto config = new StreamingToolkit::NvEncConfig();
+
+		ConfigParser::ParseNvEncConfig(baseFilePath + std::string(ConfigParser::kNvEncConfigPath), config);
+
+		return std::shared_ptr<StreamingToolkit::NvEncConfig>(config);
+	});
+}
+
+void ConfigParser::ParseWebRTCConfig(const std::string& path, StreamingToolkit::WebRTCConfig* webrtcConfig)
+{
+	std::ifstream fileStream(path);
+	Json::Reader reader;
+	Json::Value root = NULL;
+	if (fileStream.good())
+	{
+		reader.parse(fileStream, root, true);
+		webrtcConfig->ice_configuration = root.get("iceConfiguration", NULL).asString();
+		if (root.isMember("turnServer"))
+		{
+			auto turnServerNode = root.get("turnServer", NULL);
+			if (turnServerNode.isMember("uri"))
+			{
+				webrtcConfig->turn_server.uri = turnServerNode.get("uri", NULL).asString();
+			}
+
+			if (turnServerNode.isMember("providerUri"))
+			{
+				webrtcConfig->turn_server.provider_uri = turnServerNode.get("providerUri", NULL).asString();
+			}
+
+			if (turnServerNode.isMember("provider"))
```

Comment:
the ordering here is such that if both values are in place old will override the new - is that desired?

Reply:
I can fix that, it certainly seems that the newer version should get priority.